### PR TITLE
Update com_google_absl: 20211102.0 (current: 17c954d)

### LINKED ISSUE #18972: Newer release available `com_google_absl`: 20211102.0 (current: 17c954d)
Package Name: com_google_absl
Current Version: 17c954d@2021-06-03 21:21:06
Available Version: 20211102.0@2021-11-03 15:26:14
Upstream releases: https://github.com/abseil/abseil-cpp/releases

### PR DESCRIPTION
**Pull request by kfaseela**
Fixes #18972

Signed-off-by: Faseela K <faseela.k@est.tech>

### CODE REVIEW

**Comment by phlax** [comment]
seems like the failures are real on this one - the error seems to be

```console
ERROR: /build/tmp/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/external/com_github_google_tcmalloc/tcmalloc/BUILD:290:11: in cc_library rule @com_github_google_tcmalloc//tcmalloc:***@com_google_absl//absl/debugging:debugging_internal' is not visible from target '@com_github_google_tcmalloc//tcmalloc:common'. Check the visibility declaration of the former target if you think the dependency is legitimate
```

related upstream issue seems to be https://github.com/google/tcmalloc/issues/103

which is resolved, so not sure if we need to update something else to fix, or not yet released

**Comment by phlax** [comment]
current tcmalloc version is 9f385356c34d4fc11f76a000b609e2b446c20667 so might be worth looking at that

**Comment by phlax** [comment]
i think you probs need to rebase to main - it sometimes does this with merging

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -105,12 +105,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Abseil",
         project_desc = "Open source collection of C++ libraries drawn from the most fundamental pieces of Google’s internal codebase",
         project_url = "https://abseil.io/",
-        version = "17c954d90d5661e27db8fc5f086085690a8372d9",
-        sha256 = "2e4ace2ed32a4ccfd29e856ad72b4fd1eae2ec060d3ba8646857fa170d6e8269",
+        version = "20211102.0",
```
Comment (wrowe) [inline bazel/repository_locations.bzl:108]: This change includes serious regressions against msvc, which is required for consideration of adoption. You can review the azure build log for details. Not only should it be -std:c99 on windows vs -std=c99, but there are significant clib deviations that absl made contrary assumptions about. In our experience MS has a slightly different impl where the spec was left ambiguous. Does not lgtm, deps change cannot be accepted.

**Review by wrowe — CHANGES_REQUESTED** [review]
Breakage noted above needs to be addressed upstream, or faulty envoy assumptions corrected within this PR.

File: bazel/repository_locations.bzl (hunk 1 above)
Comment (phlax) [reply bazel/repository_locations.bzl:108]: @wrowe any ideas on ways forward on this, grepping etc im not seeing where this flag is set, but i guess there are further issues anyway

im assuming that we do want to update, at least at some point, maybe that is incorrect

looking at the error log, it seems to be something to do with quic, not sure if that is worth looking at

**Pull request closed without merging**